### PR TITLE
Test branch

### PR DESCRIPTION
**Pull request by tvleest**
As a developer
I want to test all of my code
so that I'm sure that it works properly

### CODE REVIEW

File: TU-Agent/buildConstruction.test2g [hunk 1]
```diff
@@ -3,18 +3,24 @@ use agentKnowledge as knowledge.
 use agentGoals as goals.
 use buildConstruction as module.
 
-test tygronagent with
+test buildConstruction with
 	%check if there is at least an oldbuilding believe.
-	pre {true}
+	pre { bel(availableLandPolygon(MP)) }
```
Comment (tvleest) [inline TU-Agent/buildConstruction.test2g]: The module can still run when theres no such believe. See the code in the module.

File: TU-Agent/buildConstruction.test2g [hunk 2]
```diff
@@ -3,18 +3,24 @@ use agentKnowledge as knowledge.
...
 	in {
 		%these actions should never be called within the demolish module
 		never done(building_plan_demolish(_)).
 		never done(map_buy_land(_, _)).
 		never done(map_sell_land(_, _, _)).
 		never done(building_plan_upgrade(_, _)).
+		
+		always done(building_plan_construction(_, _, MP)), not(bel(availableLandPolygon(MP))).
```
Comment (tvleest) [inline TU-Agent/buildConstruction.test2g]: Always building when there's no land?

File: TU-Agent/buildConstruction.test2g [hunk 3]
```diff
@@ -3,18 +3,24 @@ use agentKnowledge as knowledge.
...
+		always done(building_plan_construction(_, _, _)).
```
Comment (tvleest) [inline TU-Agent/buildConstruction.test2g]: what if there is no land?

File: TU-Agent/demolish.test2g [hunk 4]
```diff
@@ -5,19 +5,30 @@ use demolish as module.
...
 		never done(building_plan_construction(_, _, _)).
 		never done(map_buy_land(_, _)).
 		never done(map_sell_land(_, _, _)).
 		never done(building_plan_upgrade(_, _)).
+		
+		always done(building_plan_demolish(_)).
+		
+		always done(building_plan_demolish(X)), bel(oldbuildings([building(X,A,B,C,D,E,F,G)|Tail])).
+		
+		never goal(createLandToBuild), bel(oldbuildings(Tail)).
```
Comment (tvleest) [inline TU-Agent/demolish.test2g]: There can be a goal to createland and a list of oldbuildings.

File: TU-Agent/demolish.test2g [hunk 5]
```diff
@@ -5,19 +5,30 @@ use demolish as module.
...
+		
+		never done(building_plan_demolish(_)), not(goal(createLandToBuild)).
+		
+		goal(createLandToBuild) leadsto done(building_plan_demolish(_)).
+		
 	}
-	post {bel(oldbuildings(X))}
+	post {bel(oldbuildings(Tail), availableLandPolygon(G))}
```
Comment (tvleest) [inline TU-Agent/demolish.test2g]: Tail has no relation to the Tail variable in the pre and the in part.
Same goes for G, might as well use an underscore here.

File: TU-Agent/buildConstruction.test2g (hunk 2 above)
Comment (MichielvdBerg) [inline TU-Agent/buildConstruction.test2g]: If you are in the buildConstruction module, you always build right?
If you don't want to build, you don't want to go to the build module

Comment (MichielvdBerg) [inline TU-Agent/buildConstruction.test2g]: You are right, I misunderstood

**Comment by tvleest** [comment]
I'm going to refactor some tests to follow the templates in the programming guide.

**Comment by tvleest** [comment]
It passes locally with the exact same goal jar......

**Comment by tvleest** [comment]
The good news is that the tests are working in travis.
The bad news is that the in{} test in the test.test2g file does not pass whatever is in it.
Since the other modules do get tested properly I do suggest that we merge this.

As a developer
I want to test all of my code
so that I'm sure that it works properly

**Comment by tvleest** [comment]
The good news is that the tests are working in travis.
The bad news is that the in{} test in the test.test2g file does not pass whatever is in it.
Since the other modules do get tested properly I do suggest that we merge this.

As a developer
I want to test all of my code
so that I'm sure that it works properly

**Comment by jzoon** [comment]
Tests succeed, so approved

File: TU-Agent/test.test2g [hunk 6]
```diff
@@ -2,17 +2,17 @@ use TUAgent as mas.
...
+%We're running into very strange problems with this file in travis
+test tygron with
+	pre {goal(indicatorgoal(_, _))}
 	in {
-		never done(building_plan_construction(_, _, _)).
-		never done(map_buy_land(_, _)).
+		%apparently travis doesn't allow us to test the in clauses
 	}
-	post {true}
+ 	post { }
```
Comment (MichielvdBerg) [inline TU-Agent/test.test2g]: The post condition is empty, shouldn't this be post{ true }?

Comment (tvleest) [inline TU-Agent/test.test2g]: Fixed it.

**Comment by MichielvdBerg** [comment]
Approved.

**Pull request merged**